dll/win32/kernel32/kernel32_vista/vista.c

```diff
@@ -89,10 +115,12 @@ QueryFullProcessImageNameW(HANDLE hProcess,
  */
 BOOL
 WINAPI
-QueryFullProcessImageNameA(HANDLE hProcess,
-                           DWORD dwFlags,
-                           LPSTR lpExeName,
-                           PDWORD pdwSize)
+QueryFullProcessImageNameA(
+    _In_ HANDLE hProcess,
+    _In_ DWORD dwFlags,
+    _Out_z_cap_post_count_(*pdwSize, *pdwSize + 1)
+        LPSTR lpExeName,
+    _Inout_ PDWORD pdwSize)
```

Comment:
Note: in the WideCharToMultiByte call below, it would be safer to use (for the 4th parameter), `*pdwSize + 1` instead of assuming (with `-1`) the returned string is NULL-terminated (even if it normally is).

The `strlen(lpExeName);` call that follows it is also not optimal and could be replaced by the returned value of the WideCharToMultiByte call.